org-formation/090-systems-manager/ScheduledMaintenance.yaml

```diff
@@ -0,0 +1,127 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: >-
+  AWS CloudFormation template that will create an SSM Maintenance
+  Window Task, with associated Maintenance Window, SSM Document,
+  Target and Role to run a given script on the EC2s in the specified account.
+Parameters:
+  Schedule:
+    Type: 'String'
+    Description: 'Schedule to run script'
+    Default: "rate(24 hours)"
+  ScriptURL:
+    Type: 'String'
+    Description: 'URL for the script to run'
+  ScriptParameters:
+    Type: 'String'
+    Description: >-
+      space delimited list of key:value, where key is an SSM Parameter Store
+      key and value is the shell envvar name to assign.
+Resources:
+  # Create IAM Role for SSM
+  SSMExecutionRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: "ssm.amazonaws.com"
+            Action: "sts:AssumeRole"
+      Policies:
+        - PolicyName: "SSMExecutionPolicy"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "ec2:DescribeInstances"
+                  - "ssm:SendCommand"
+                Resource: "*"
+        - PolicyName: "SSMParameterPolicy"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "ssm:GetParameters"
+                Resource:
+                - !Join [ '', ['arn:', !Ref 'AWS::Partition', ':ssm:', !Ref 'AWS::Region', ':', !Ref 'AWS::AccountId', '*'] ]
+  SSMDocument:
+    Type: "AWS::SSM::Document"
+    Properties:
+      DocumentType: "Command"
+      Content:
+        schemaVersion: "2.2"
+        description: "Run a script on EC2 instances"
+        parameters:
+          ScriptUrl:
+            type: String
+            description: "The URL of the script to run"
+          Region:
+            type: String
+            description: The AWS region
+          SsmParameterNameToEnvvarMap:
+            type: String
+            description: >-
+              space delimited list of key:value, where key is an SSM Parameter Store
+              key and value is the shell envvar name to assign.
+        mainSteps:
+        - action: "aws:runShellScript"
+          name: "runShellScript"
+          inputs:
+            runCommand:
+            - SSM_TO_ENV_MAP=( {{ SsmParameterNameToEnvvarMap }} )
```

Comment:
what does `SsmParameterNameToEnvvarMap` look like?

Reply:
I added an example to the parameter 'description'.